app/mailers/spree/user_mailer.rb

```diff
@@ -1,7 +1,7 @@
 module Spree
   class UserMailer < BaseMailer
     def reset_password_instructions(user, token, *args)
-      @edit_password_reset_url = spree.edit_spree_user_password_url(:reset_password_token => token)
+      @edit_password_reset_url = spree.edit_spree_user_password_url(:reset_password_token => token, :host => Spree::Config.site_url)
```

Comment:
The need for this change doesn't sit right with me. Using `spree.edit_spree_user_password_url` should form a fully qualified url based on the request that triggered this code. How is the code getting demo.spreecommerce.com as the host, unless you're actually hitting the page at that url? There must be a configuration somewhere that is setting this value. Perhaps this is working as designed and your site just didn't have the value configured?